src/checkpoint/checkpoint_meta.c

```diff
@@ -0,0 +1,74 @@
+/*-
+ * Copyright (c) 2014-present MongoDB, Inc.
+ * Copyright (c) 2008-2014 WiredTiger, Inc.
+ *	All rights reserved.
+ *
+ * See the file LICENSE for redistribution information.
+ */
+
+#include "wt_internal.h"
+
+/*
+ * __wt_checkpoint_ckptlist_free --
+ *     Discard the checkpoint array.
+ */
+void
+__wt_checkpoint_ckptlist_free(WT_SESSION_IMPL *session, WT_CKPT **ckptbasep)
+{
+    WT_CKPT *ckpt, *ckptbase;
+
+    if ((ckptbase = *ckptbasep) == NULL)
+        return;
+
+    /*
+     * Sometimes the checkpoint list has a checkpoint which has not been named yet, but carries an
+     * order number.
+     */
+    WT_CKPT_FOREACH_NAME_OR_ORDER (ckptbase, ckpt)
+        __wt_checkpoint_free(session, ckpt);
+    __wt_free(session, *ckptbasep);
+}
+
+/*
+ * __wt_checkpoint_saved_ckptlist_free --
+ *     Discard the saved checkpoint list.
+ */
+void
+__wt_checkpoint_saved_ckptlist_free(WT_SESSION_IMPL *session)
+{
+    WT_BTREE *btree;
+
+    btree = S2BT(session);
+
+    __wt_checkpoint_ckptlist_free(session, &btree->ckpt);
+    btree->ckpt_bytes_allocated = 0;
```

Comment:
This annoys me a bit that we have `btree` stuff here.

Reply:
Fair enough, but it's no worse than the cross module of meta/btree before